cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/workflow/DefaultWorkflowConditionConfigurer.java

```diff
@@ -42,8 +45,10 @@
   private List<WorkflowNode> currentBranch;
   private boolean addingToIfBranch = true;
   private final String predicateClassName;
+  private final String uniqueName;
 
-  public DefaultWorkflowConditionConfigurer(T parentConfigurer, String predicateClassName) {
+  public DefaultWorkflowConditionConfigurer(String uniqueName, T parentConfigurer, String predicateClassName) {
```

Comment:
Since the `uniqueName` is the identifier for the node and will be non-null its better to have it as first parameter.